### PR TITLE
Added some options for choosing Cloud SQL generation and the db region.

### LINKED ISSUE #68: [App Engine] Wordpress Standard Environment - Functionality to use CloudSQL 2nd gen
When running the WordPress code generation, It does not let you populate the CloudSQL connection name with the region. I have created a fork with a rough implantation of this feature. 

Would there be an interest to add the functionality to choose between using 1st or 2nd generation CloudSQL instance?

Current example behavior:
```
:/cloudsql/{{project_id}}:{{db_instance}}
```

Expected example behavior:
```
:/cloudsql/{{project_id}}:{{db_region}}:{{db_instance}}
```

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #68

### CODE REVIEW

**Comment by tmatsuo** [comment]
The WordPress tests are passing on our jenkins :)

File: appengine/wordpress/src/WordPressSetup.php [hunk 1]
```diff
@@ -64,13 +76,29 @@ protected function configure()
                 self::DEFAULT_DIR
             )
             ->addOption(
+                'sql_gen',
+                '',
+                InputOption::VALUE_OPTIONAL,
+                'Cloud SQL generation to use; 2: '
+                . self::CLOUD_SQL_SECOND_GEN . ', 1: '
+                . self::CLOUD_SQL_FIRST_GEN . '.',
+                null
```
Comment (bshaffer) [inline appengine/wordpress/src/WordPressSetup.php]: we should actually set the default to `2`, so this is surfaced in the documentation.

File: appengine/wordpress/src/WordPressSetup.php [hunk 2]
```diff
@@ -225,15 +253,43 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+        if ($sql_gen === '1') {
+            if ($env === self::FLEXIBLE_ENV) {
+                $output->writeln('<error>You can not use '
+                                 . self::CLOUD_SQL_FIRST_GEN . ' with '
+                                 . self::FLEXIBLE_ENV . '.</error>');
+                return self::DEFAULT_ERROR;
+            }
+            $sql_gen = self::CLOUD_SQL_FIRST_GEN;
+            $db_region_str = ':';
+        } else {
```
Comment (bshaffer) [inline appengine/wordpress/src/WordPressSetup.php]: How about a `switch` statement with an exception for invalid `sql_gen`?

File: appengine/wordpress/src/WordPressSetup.php [hunk 3]
```diff
@@ -64,13 +76,29 @@ protected function configure()
                 self::DEFAULT_DIR
             )
             ->addOption(
+                'sql_gen',
+                '',
+                InputOption::VALUE_OPTIONAL,
+                'Cloud SQL generation to use; 2: '
```
Comment (bshaffer) [inline appengine/wordpress/src/WordPressSetup.php]: would definitely rather see this as 

```php
sprintf('Cloud SQL generation to use; 2: %s, 1:%s',
```

File: appengine/wordpress/src/WordPressSetup.php [hunk 4]
```diff
@@ -225,15 +253,43 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-
+        $sql_gen = $input->getOption('sql_gen');
+        if ($sql_gen === '1') {
+            if ($env === self::FLEXIBLE_ENV) {
+                $output->writeln('<error>You can not use '
+                                 . self::CLOUD_SQL_FIRST_GEN . ' with '
+                                 . self::FLEXIBLE_ENV . '.</error>');
+                return self::DEFAULT_ERROR;
+            }
+            $sql_gen = self::CLOUD_SQL_FIRST_GEN;
```
Comment (bshaffer) [inline appengine/wordpress/src/WordPressSetup.php]: Why do we set `$sql_gen` to this? We do not use it again.

File: appengine/wordpress/src/WordPressSetup.php [hunk 5]
```diff
@@ -40,6 +40,18 @@ class WordPressSetup extends Command
     const FLEXIBLE_ENV = 'Flexible Environment';
     const STANDARD_ENV = 'Standard Environment';
 
+    const CLOUD_SQL_FIRST_GEN = 'Cloud SQL First Generation';
+    const CLOUD_SQL_SECOND_GEN = 'Cloud SQL Second Generation';
```
Comment (bshaffer) [inline appengine/wordpress/src/WordPressSetup.php]: Not a huge fan of using constants for interface strings. I do not see an advantage to doing this.

File: appengine/wordpress/src/WordPressSetup.php (hunk 1 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Done

File: appengine/wordpress/src/WordPressSetup.php (hunk 2 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Done

File: appengine/wordpress/src/WordPressSetup.php (hunk 3 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Done

File: appengine/wordpress/src/WordPressSetup.php (hunk 5 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Good catch, done

File: appengine/wordpress/src/WordPressSetup.php (hunk 4 above)
Comment (tmatsuo) [inline appengine/wordpress/src/WordPressSetup.php]: Good catch, done

**Comment by tmatsuo** [comment]
@bshaffer Ready for another look, PTAL

**Comment by tmatsuo** [comment]
FYI, I just moved sql_gen part upper than downloading wordpress

**Comment by tmatsuo** [comment]
@bshaffer The tests are passing on our jenkins too :)

File: appengine/wordpress/src/files/flexible/app.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@ runtime: php
 vm: true
 
 beta_settings:
-  cloud_sql_instances: {{project_id}}:us-central1:{{db_instance}}
+  cloud_sql_instances: {{project_id}}{{db_region}}{{db_instance}}
```
Comment (bshaffer) [inline appengine/wordpress/src/files/flexible/app.yaml]: I would rather be explicit with the Instance Connection Name between 1st and 2nd gen. Something like:
```php
switch ($sql_gen) {
    case '1':
        // ...
        $connection_name = sprintf('%s:%s', $project_id, $db_instance);
    case '2':
        // ...
        $connection_name = sprintf('%s:%s:%s', $project_id, $db_region, $db_instance);
```
and then in `app.yaml` / `DB_HOST`:

```yaml
beta_settings:
    cloud_sql_instances: {{connection_name}}
```

Comment (tmatsuo) [inline appengine/wordpress/src/files/flexible/app.yaml]: Yeah, I agree, but it's kind of complicated.

I want to fail fast (before downloading WordPress) so I want to check $sql_gen before that. On the other hand, when I check the environment (flexible/standard) we only download the app engine standard, but at that time, wordpress should be there, etc.

Anyways, point taken, I changed it to create a pattern for sprintf.

Comment (bshaffer) [inline appengine/wordpress/src/files/flexible/app.yaml]: Your changes work for me! It's much more clear what's going on. Thank you!

**Comment by bshaffer** [comment]
image

**Pull request merged**